src/app-middlewares/after/callback-status-catcher.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+let ensureOutputEnvelope = require('../../tools/envelope').ensureOutputEnvelope;
+let STATUSES = require('../../constants').STATUSES;
+const _ = require('lodash');
+/*
+this method creates the correct envelope responses if the app has used a callback url in their code
+by signalling to Zapier that this app/method is returning a callback status the task will be placed
+in a waiting state until the callback is called.
+*/
+let callbackStatusCatcher = output => {
+  const input = output.input || {};
+
+  const callbackUsed = _.get(input, '_zapier.event.callbackUsed');
+  if (callbackUsed) {
+    output = ensureOutputEnvelope(output);
+    output.status = STATUSES.CALLBACK;
```

Comment:
Trying to think through how this will interact with [this code](https://github.com/zapier/zapier-platform-core/blob/d9edbf30546b1597a61b6119c4a558195b013de8/src/middleware.js#L49-L55) and having a bit of a hard time wrapping my head around it. Are we safe to assume output will be an envelope at this point?

Otherwise do we need to be concerned about the flag to skip envelope. Also, would that mean that depending on use of `getCallbackUrl()`, any app middlewares that come after this will potentially have two different structures to be concerned with?

Hoping that we can assume it's an envelop and that way it's consistent throughout.

Reply:
`ensureOutputEnvelope` can be called multiple times as it'll realise it's an envelope shape already and `ensureOutputEnvelope` becomes a no-op.
https://github.com/zapier/zapier-platform-core/blob/d9edbf30546b1597a61b6119c4a558195b013de8/src/middleware.js#L91

says that ensureEnvelope is called before processing middleware, so I think you may be correct that callback-status-catcher should perhaps ensure that options. skipEnvelope isn't set. When is that `true`? 

We already have this case that if `options. skipEnvelope` is `true` your middleware will be called with just the results but if it's false it'll be called with an envelope